From d56cf122d12632a0015e19778a71c88f80799227 Mon Sep 17 00:00:00 2001
From: Stewart Smith <stewart@linux.vnet.ibm.com>
Date: Sat, 3 Oct 2015 11:02:31 +1000
Subject: [PATCH 1/2] Fix crash in ast fbcon on boot

theory is ops->cur_blink_jiffies is not getting initialised before adding
the timer

Reported-by: Stephanie M Swanson <swanman@us.ibm.com>
Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
Signed-off-by: Stewart Smith <stewart@linux.vnet.ibm.com>
---
 drivers/video/console/fbcon.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/drivers/video/console/fbcon.c b/drivers/video/console/fbcon.c
index 1aaf893..45d2a0a 100644
--- a/drivers/video/console/fbcon.c
+++ b/drivers/video/console/fbcon.c
@@ -416,6 +416,8 @@ static void fbcon_add_cursor_timer(struct fb_info *info)
 			INIT_WORK(&info->queue, fb_flashcursor);
 
 		init_timer(&ops->cursor_timer);
+		if (ops->cur_blink_jiffies < 1)
+			ops->cur_blink_jiffies = msecs_to_jiffies(200);
 		ops->cursor_timer.function = cursor_timer_handler;
 		ops->cursor_timer.expires = jiffies + ops->cur_blink_jiffies;
 		ops->cursor_timer.data = (unsigned long ) info;
-- 
1.7.1

